[AArch64] Fix SVE scalar fcopysign lowering without neon.

davemgreen opened the pull request:
Without this we can try to generate invalid instructions or create illegal types. This patch generates a SVE fcopysign instead and use its lowering. BF16 is left out of the moment as it doesn't lower successfully (but could use the same code as fp16).

paulwalker-arm commented:
Just a FYI but I have recently refactored LowerFCOPYSIGN to simplify it somewhat whilst at the same time making it work better for bfloat types.  Nothing to block this PR, but you can leave the bfloat side of things to me.

davemgreen committed:
[AArch64] Fix SVE scalar fcopysign lowering without neon.

Without this we can try to generate invalid instructions or create illegal
types. Generate a SVE fcopysign instead and use its lowering. BF16 is left out
of the moment as it doesn't lower successfully (but could use the same code as
fp16).

david-arm reviewed (approved):
LGTM!

davemgreen commented:
/cherry-pick d4ab3df320f9eebf11cc5fb600a0919f93678abe

davemgreen commented:
/cherry-pick 4c2d1b4c53de d4ab3df320f9
